### PR TITLE
update(settings): update keybinds functionality AP-174

### PR DESCRIPTION
**Pull request by davidchan1219**
**What this PR does** 📖

Added the system/browser hotkey check feature in keybind function

**Which issue(s) this PR fixes** 🔨

AP-174

### CODE REVIEW

File: components/views/settings/pages/keybinds/index.vue [hunk 1]
```diff
@@ -147,7 +152,10 @@ export default Vue.extend({
       const blockedChars = ['capslock', 'delete']
       const hasBlockedChars = blockedChars.includes(key)
 
-      if (keyAlreadyBound) {
+      if (keyAlreadyExist) {
+        this.$data.editingKeybind.error = true
+        this.$data.editingKeybind.errorMessage = 'That is browser/system shortcut. Please input other keys.'
```
Comment (Jekrimo) [inline components/views/settings/pages/keybinds/index.vue:157]: Will you add this to our translations file?

File: components/views/settings/pages/keybinds/index.vue [hunk 2]
```diff
@@ -147,7 +152,10 @@ export default Vue.extend({
...
+      } else if (keyAlreadyBound) {
         this.$data.editingKeybind.error = true
         this.$data.editingKeybind.errorMessage = 'Key already bound'
```
Comment (Jekrimo) [inline components/views/settings/pages/keybinds/index.vue:160]: Here too

File: components/views/settings/pages/keybinds/index.vue [hunk 3]
```diff
@@ -135,6 +136,10 @@ export default Vue.extend({
       const newString = this.$data.editingKeybind.newString
       const keyAlreadyBound = newString.includes(key)
 
+      const keyAlreadyExist = navigator.userAgent.indexOf("Mac") ? 
+        macShortcuts.includes(newString + '+' + key) : 
+        windowsShortcuts.includes(newString + '+' + key)
+
       const modifiers = ['shift', 'control', 'alt', 'meta', 'tab', 'capslock']
```
Comment (Jekrimo) [inline components/views/settings/pages/keybinds/index.vue:143]: I know you didn't write these but will you make enum for these and the blockedChars as well? located in libraries/Enums/types then export through the enums.ts file

**Review by Jekrimo — CHANGES_REQUESTED** [review]

**Review by Jekrimo — APPROVED** [review]

**Comment by Jekrimo** [comment]
> 

Approved, thank you! Will you do one last thing and fill out the pr template for us?

File: components/views/settings/pages/keybinds/index.vue [hunk 4]
```diff
@@ -144,24 +163,31 @@ export default Vue.extend({
...
+        this.$data.editingKeybind.errorMessage = this.$t('pages.settings.keybinds.existHotkeyError')
       } else if (modifierAfterAlphanumeric) {
         this.$data.editingKeybind.error = true
-        this.$data.editingKeybind.errorMessage =
-          'Modifiers (Shift, Tab, Option, etc.) Must Come Before Alphanumerics'
+        this.$data.editingKeybind.errorMessage = this.$t('pages.settings.keybinds.modifierHotkeyError')
       } else if (hasBlockedChars) {
         this.$data.editingKeybind.error = true
-        this.$data.editingKeybind.errorMessage = 'Character Not Allowed'
+        this.$data.editingKeybind.errorMessage = this.$t('pages.settings.keybinds.editHotkeyError')
```
Comment (WanderingHogan) [inline components/views/settings/pages/keybinds/index.vue:180]: thank you for also adding translations!

**Review by WanderingHogan — APPROVED** [review]
this is great, thank you @davidchan1219

**Comment by stavares843** [comment]
Captura de ecrã 2022-01-08, às 00 05 57

im using macOS, should this say ''optionKey'' instead of ''alt''?

**Comment by stavares843** [comment]
Captura de ecrã 2022-01-08, às 00 04 33

we could add a bit more padding between the error and the field above

also, should the error be red as the rest of the error messages?

**Comment by stavares843** [comment]
image

when I go to the chat and I do optionKey (same as alt for macOS) + s, doesn't open the settings

**Review by stavares843 — CHANGES_REQUESTED** [review]
added couple comments

**Comment by WanderingHogan** [comment]
Good ideas, @stavares843 :

option instead of alt if you detect it's a mac is a good idea. Can you make:
meta = 'Command'
alt = 'Option'

More spacing and an error color are also good.

I think the third thing you mentioned might be kind of a big thing and I broke it out into another ticket: AP-427

**Comment by stavares843** [comment]
thanks for checking! 🎉

**Comment by WanderingHogan** [comment]
@davidchan1219 please check https://github.com/Satellite-im/Core-PWA/pull/976 when you get a chance, and make sure it has everything you want for this PR, then close this if it does

**Comment by davidchan1219** [comment]
> @davidchan1219 please check #976 when you get a chance, and make sure it has everything you want for this PR, then close this if it does

In the new branch, everything works fine.

**Pull request closed without merging**